src/field_specials.c

```diff
@@ -2018,130 +2058,191 @@ bool32 sub_8139ED0(void)
-        { BattleFrontier_Lounge2_Text_26174D, BattleFrontier_Lounge2_Text_2617F9, BattleFrontier_Lounge2_Text_2618C4 },
+    *var += daysSince;
+    *var %= FRONTIER_MANIAC_FACILITY_COUNT;
+}
+
+void ShowFrontierManiacMessage(void)
+{
+    static const u8 *const sFrontierManiacMessages[][FRONTIER_MANIAC_MESSAGE_COUNT] = 
+    {
+        [FRONTIER_MANIAC_BATTLE_TOWER_SINGLES] =
+        { 
+            BattleFrontier_Lounge2_Text_260971, 
+            BattleFrontier_Lounge2_Text_260A1E, 
+            BattleFrontier_Lounge2_Text_260AE7 
+        },
+        [FRONTIER_MANIAC_BATTLE_TOWER_DOUBLES] =
+        { 
+            BattleFrontier_Lounge2_Text_2619AC, 
+            BattleFrontier_Lounge2_Text_261A91, 
+            BattleFrontier_Lounge2_Text_261B0C 
+        },
+        [FRONTIER_MANIAC_BATTLE_TOWER_MULTIS] = 
+        { 
+            BattleFrontier_Lounge2_Text_261B95, 
+            BattleFrontier_Lounge2_Text_261B95, 
+            BattleFrontier_Lounge2_Text_261B95 
+        },
+        [FRONTIER_MANIAC_BATTLE_TOWER_LINK_MULTIS] =
+        { 
+            BattleFrontier_Lounge2_Text_261C1A, 
+            BattleFrontier_Lounge2_Text_261C1A, 
+            BattleFrontier_Lounge2_Text_261C1A 
+        },
+        [FRONTIER_MANIAC_BATTLE_DOME] =
+        { 
+            BattleFrontier_Lounge2_Text_260BC4, 
+            BattleFrontier_Lounge2_Text_260C6D, 
+            BattleFrontier_Lounge2_Text_260D3A 
+        },
+        [FRONTIER_MANIAC_BATTLE_FACTORY] =
+        { 
+            BattleFrontier_Lounge2_Text_260E1E, 
+            BattleFrontier_Lounge2_Text_260EC7, 
+            BattleFrontier_Lounge2_Text_260F74 
+        },
+        [FRONTIER_MANIAC_BATTLE_PALACE] =
+        { 
+            BattleFrontier_Lounge2_Text_2614E6, 
+            BattleFrontier_Lounge2_Text_261591, 
+            BattleFrontier_Lounge2_Text_26166F 
+        },
+        [FRONTIER_MANIAC_BATTLE_ARENA] =
+        { 
+            BattleFrontier_Lounge2_Text_261282, 
+            BattleFrontier_Lounge2_Text_261329, 
+            BattleFrontier_Lounge2_Text_261403 
+        },
+        [FRONTIER_MANIAC_BATTLE_PIKE] = 
+        { 
+            BattleFrontier_Lounge2_Text_261026, 
+            BattleFrontier_Lounge2_Text_2610CC, 
+            BattleFrontier_Lounge2_Text_261194 
+        },
+        [FRONTIER_MANIAC_BATTLE_PYRAMID] =
+        { 
+            BattleFrontier_Lounge2_Text_26174D, 
+            BattleFrontier_Lounge2_Text_2617F9, 
+            BattleFrontier_Lounge2_Text_2618C4 
+        },
     };
 
-    static const u8 gUnknown_085B2CC8[][2] = {
-        { 0x15, 0x38 },
-        { 0x15, 0x23 },
-        { 0xff, 0xff },
-        { 0xff, 0xff },
-        { 0x02, 0x04 },
-        { 0x07, 0x15 },
-        { 0x07, 0x15 },
-        { 0x0e, 0x1c },
-        { 0x0d, 0x70 },
-        { 0x07, 0x38 }
+    static const u8 sFrontierManiacStreakThresholds[][FRONTIER_MANIAC_MESSAGE_COUNT - 1] = 
```

Comment:
`FRONTIER_MANIAC_MESSAGE_COUNT - 1` doesn't make sense here.  This data is independent from the `sFrontierManiacMessages` data.

Reply:
The messages in `sFrontierManiacMessages` are printed depending on whether or not the streaks in `sFrontierManiacStreakThresholds` have been achieved. So if the player has a streak `> sFrontierManiacStreakThresholds[facility][1]`, then `sFrontierManiacMessages[facility][2]` is shown, if they have a streak `> sFrontierManiacStreakThresholds[facility][0]` but `<= sFrontierManiacStreakThresholds[facility][1]` then `sFrontierManiacMessages[facility][1]` is shown, and if they have a streak `<= sFrontierManiacStreakThresholds[facility][0]` then `sFrontierManiacMessages[facility][0]` is shown. This is demonstrated by the last two lines

```for (i = 0; i < FRONTIER_MANIAC_MESSAGE_COUNT - 1 && sFrontierManiacStreakThresholds[facility][i] < winStreak; i++);

ShowFieldMessage(sFrontierManiacMessages[facility][i]);```

Maybe the relationship would be clearer if something like `#define FRONTIER_MANIAC_STREAK_COUNT   FRONTIER_MANIAC_MESSAGE_COUNT - 1` was added.